src/System.Net.Http.WinHttpHandler/src/System/Net/Http/CertificateHelper.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Security;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net.Http
+{
+    // TODO: Move to System.Net.Common as part of System.Private.Networking.dll refactor.
+    internal static class CertificateHelper
+    {
+        public static void BuildChain(
+            X509Certificate2 certificate,
+            string hostName,
+            bool checkCertificateRevocationList,
+            out X509Chain chain,
+            out SslPolicyErrors sslPolicyErrors)
+        {
+            chain = null;
+            sslPolicyErrors = SslPolicyErrors.None;
+
+            // Build the chain.
+            chain = new X509Chain();
+            chain.ChainPolicy.RevocationMode =
+                checkCertificateRevocationList ? X509RevocationMode.Online : X509RevocationMode.NoCheck;
+            chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
+            chain.Build(certificate);
+            if (chain.ChainStatus != null && chain.ChainStatus.Length != 0)
+            {
+                sslPolicyErrors |= SslPolicyErrors.RemoteCertificateChainErrors;
+            }
```

Comment:
In theory this if block is performing the same check as the return value of `chain.Build(certificate)` (which is being ignored)...